### PR TITLE
Fixed the back-end error when tried to getting the user information.

### LINKED ISSUE #1: Observed back-end error  when tried to getting the user information.
**Description:**
Got the following back-end error  when tried to getting the user information.  

" ERROR {org.wso2.carbon.identity.authenticator.dropbox.DropboxAuthenticator} -  Error occurred while accessing user info endpoint
java.io.IOException: Server returned HTTP response code: 400 for URL: https://api.dropboxapi.com/1/account/info
"

**Suggested Labels:**

**Suggested Assignees:**

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**

**Related Issues:**

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
Fixed #1.

## Goals
 changed the all dropbox's requirements according to the API v2 . 

## Approach
N/A

## User stories
N/A

## Release note
N/A

## Documentation
N/A

## Training
N/A

## Certification
N/A

## Marketing
N/A

## Automation tests
 - Unit tests 
  N/A

 - Integration tests
N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples
N/A

## Related PRs
N/A

## Migrations (if applicable)
N/A

## Test environment
N/A
 
## Learning
N/A

### CODE REVIEW

File: component/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java [hunk 1]
```diff
@@ -216,4 +222,42 @@ private OAuthClientRequest getAccessRequest(String tokenEndPoint, String clientI
...
+     * @return Response string.
+     * @throws IOException
+     */
+    protected String sendRequest(String url, String accessToken)
+            throws IOException {
+        if (log.isDebugEnabled()) {
+            log.debug("Claim URL: " + url);
+        }
+
+        if (url == null) {
```
Comment (Kanapriya) [inline component/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java]: Use StringUtils for the null check

File: component/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java [hunk 2]
```diff
@@ -216,4 +222,42 @@ private OAuthClientRequest getAccessRequest(String tokenEndPoint, String clientI
...
+    /**
+     * Request user claims from user info endpoint.
+     *
+     * @param url         User info endpoint.
+     * @param accessToken Access token.
+     * @return Response string.
+     * @throws IOException
+     */
+    protected String sendRequest(String url, String accessToken)
+            throws IOException {
```
Comment (Kanapriya) [inline component/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java]: Formatting issue

File: component/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java [hunk 3]
```diff
@@ -216,4 +222,42 @@ private OAuthClientRequest getAccessRequest(String tokenEndPoint, String clientI
...
+        }
+
+        URL obj = new URL(url);
+        HttpURLConnection urlConnection = (HttpURLConnection) obj.openConnection();
+        urlConnection.setRequestMethod("POST");
+        urlConnection.setRequestProperty("Authorization", "Bearer " + accessToken);
+        BufferedReader reader = new BufferedReader(new InputStreamReader(urlConnection.getInputStream()));
+        StringBuilder builder = new StringBuilder();
+        String inputLine = reader.readLine();
+        while (inputLine != null) {
```
Comment (Kanapriya) [inline component/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java]: Use StringUtils for null check

File: component/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java [hunk 4]
```diff
@@ -216,4 +222,42 @@ private OAuthClientRequest getAccessRequest(String tokenEndPoint, String clientI
...
+            builder.append(inputLine).append("\n");
+            inputLine = reader.readLine();
+        }
+        reader.close();
+
+        if (log.isDebugEnabled() && IdentityUtil.isTokenLoggable(
+                org.wso2.carbon.identity.base.IdentityConstants.IdentityTokens.USER_ID_TOKEN)) {
+            log.debug("response: " + builder.toString());
```
Comment (Kanapriya) [inline component/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java]: Please add meaningful message to the debug logs

**Review by Kanapriya — CHANGES_REQUESTED** [review]

File: component/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java [hunk 5]
```diff
@@ -216,4 +223,39 @@ private OAuthClientRequest getAccessRequest(String tokenEndPoint, String clientI
...
+        urlConnection.setRequestMethod("POST");
+        urlConnection.setRequestProperty("Authorization", "Bearer " + accessToken);
+        BufferedReader reader = new BufferedReader(new InputStreamReader(urlConnection.getInputStream()));
+        StringBuilder builder = new StringBuilder();
+        String inputLine = reader.readLine();
+        while (StringUtils.isNotEmpty(inputLine)) {
+            builder.append(inputLine).append("\n");
+            inputLine = reader.readLine();
+        }
+        reader.close();
```
Comment (muthulee) [inline component/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java]: Please close the resource stream on finally block or use try resource

**Review by dimslee — CHANGES_REQUESTED** [review]

File: component/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java [hunk 6]
```diff
@@ -216,4 +223,44 @@ private OAuthClientRequest getAccessRequest(String tokenEndPoint, String clientI
...
+                builder.append(inputLine).append("\n");
+                inputLine = reader.readLine();
+            }
+        } finally {
+            if(reader != null) {
+                reader.close();
+            }
+        }
+        if (log.isDebugEnabled() && IdentityUtil.isTokenLoggable(IdentityTokens.USER_ID_TOKEN)) {
+            log.debug(" Dropbox user information : " + builder.toString());
```
Comment (sameeragunarathne) [inline component/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java]: Unwanted space in the start if the debug log message

File: component/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java [hunk 7]
```diff
@@ -216,4 +223,44 @@ private OAuthClientRequest getAccessRequest(String tokenEndPoint, String clientI
...
+            log.debug("Claim URL: " + url);
+        }
+
+        if (StringUtils.isEmpty(url)) {
+            return StringUtils.EMPTY;
+        }
+
+        URL obj = new URL(url);
+        HttpURLConnection urlConnection = (HttpURLConnection) obj.openConnection();
+        urlConnection.setRequestMethod("POST");
```
Comment (hasuniea) [inline component/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java]: Can't we use constants?

**Review by hasuniea — APPROVED** [review]

**Review by dimslee — APPROVED** [review]

**Review by Kanapriya — APPROVED** [review]

**Review by sameeragunarathne — APPROVED** [review]

**Review by Kanapriya — APPROVED** [review]

**Review by Biruntha — APPROVED** [review]

**Review by menakaj — COMMENTED** [review]
Please use proper commit messages. Fix in the other places as well.

**Review by thanujalk — APPROVED** [review]

**Pull request merged**